controlplane/kubeadm/internal/webhooks/kubeadmcontrolplanetemplate.go

```diff
@@ -59,8 +59,6 @@ func (webhook *KubeadmControlPlaneTemplate) Default(_ context.Context, obj runti
 
 	k.Spec.Template.Spec.KubeadmConfigSpec.Default()
 
-	k.Spec.Template.Spec.RolloutStrategy = defaultRolloutStrategy(k.Spec.Template.Spec.RolloutStrategy)
```

Comment:
I just realized that this will break folks after the CAPI update.

The problem is if someone applied a KubeadmControlPlaneTemplate without rolloutStrategy being set and keeps applying it.

The ValidateUpdate code ensures that Spec.Template.Spec is immutable.

So the pre-existing object will have rolloutStrategy set, the new one won't and this will lead to errors during what is supposed-to-be no-op updates.

I'll open a follow-up PR to fix this